tests/variants.test.css

```diff
@@ -111,17 +111,15 @@
   font-size: 1.125rem;
   line-height: 1.75rem;
 }
-.marker\:text-red-500 *::marker {
-  --tw-text-opacity: 1;
-  color: rgb(239 68 68 / var(--tw-text-opacity));
-}
 .marker\:text-lg::marker {
   font-size: 1.125rem;
   line-height: 1.75rem;
 }
+.marker\:text-red-500 *::marker {
+  color: rgb(239 68 68);
+}
```

Comment:
I think what's happening is the utilities are being grouped by utility now instead of by the selector, it's like:

```
.marker\:poop *::marker {}
.marker\:poop::marker {}

.marker\:banana *::marker {}
.marker\:banana::marker {}

.marker\:tonsils *::marker {}
.marker\:tonsils::marker {}
```

...instead of:

```
.marker\:poop *::marker {}
.marker\:banana *::marker {}
.marker\:tonsils *::marker {}

.marker\:poop::marker {}
.marker\:banana::marker {}
.marker\:tonsils::marker {}
```

I'd have to think about it harder but it feels like there was an important reason we originally did it by selector, so we probably want to be careful here. It's tricky admittedly for the reasons we talked about the other week, where variants registered within a function share a sort order or whatever.

I wonder if what we really want is to make this API possible for this one:

```
addVariant('marker', [() => {...}, () => {...}])
```

...so we can know the count up front and adjust the sort appropriately?

Reply:
Good news @adamwathan that API already works :D — I've updated the PR to switch to it